README.md

```diff
@@ -46,70 +38,172 @@ Maven:
 
 ## Usage
 
-Instantiate the task.
+To create the task instance, you need to use the `WebRTCTaskBuilder` instance
+which can be used to configure the task before creating it.
+
+The below configuration represents the default values chosen by the builder as
+if you had not configured the builder and just called `.build()` directly.
 
 ```java
-final WebRTCTask task = new WebRTCTask();
+final WebRTCTask task = new WebRTCTaskBuilder()
+    .withVersion(WebRTCTaskVersion.V1)
+    .withHandover(true)
+    .withMaxChunkLength(262144)
+    .build();
 ```
 
-Then, register a message handler:
+To send offers, answers and candidates, use the following task methods:
+
+* `task.sendOffer(offer: @NonNull Offer): void`
+* `task.sendAnswer(answer: @NonNull Answer): void`
+* `task.sendCandidates(candidate: @NonNull Candidate[]): void`
+
+You can register an event handler in the following way:
 
 ```java
 task.setMessageHandler(new MessageHandler() {
     @Override
-    public void onOffer(SessionDescription sd) {
+    public void onOffer(@NonNull Offer offer) {
         // Handle offer
     }
 
     @Override
-    public void onAnswer(SessionDescription sd) {
+    public void onAnswer(@NonNull Answer answer) {
         // Handle answer
     }
 
     @Override
-    public void onCandidates(List<IceCandidate> candidates) {
-        for (IceCandidate candidate : candidates) {
-            peerConnection.addIceCandidate(candidate);
-        }
+    public void onCandidates(@NonNull Candidate[] candidates) {
+        // Handle candidates
     }
 });
 ```
 
-Finally, pass the task instance to the `SaltyRTCBuilder` through the
-`.usingTasks(...)` method.
+### Data Channel Crypto Context
+
+The task provides another security layer for data channels which can be
+leveraged by usage of a `DataChannelCryptoContext` instance. To retrieve such
+an instance, call:
+
+```java
+final DataChannelCryptoContext context = task.getCryptoContext(dataChannel.id);
+```
+
+You can encrypt messages on the sending end in the following way:
+
+```java
+final Box box = context.encrypt(yourData);
+dataChannel.send(ByteBuffer.wrap(box.toBytes()));
+```
+
+On the receiving end, decrypt the message by the use of the crypto context:
+
+```java
+final Box box = new Box(message);
+final byte[] yourData = context.decrypt(
+    box, DataChannelCryptoContext.NONCE_LENGTH);
+```
+
+Note, that you should not use a crypto context for a data channel that is being
+used for handover. The task will take care of encryption and decryption itself.
```

Comment:
Would that even be possible? The handover is done by the client, so there are no handover messages that the user could accidentally encrypt and chunk twice, right?

Reply:
The application could create a crypto context itself before the handover took place. It can also technically use the same channel.

Reply:
Ok. I still think that for most users putting this note in the README will probably be confusing and not too relevant (since the client handles handover), but I leave it to you whether to remove it or not.

Reply:
Quick "Verständnisfrage" to make sure... Creating a crypto context simply initializes some keys, but has no side effects on the signaling channel itself, right?

Reply:
Yes. The point of the comment is just that you don't need to do crypto on the handed over signalling channel which may not be 100% clear.